javatests/arcs/core/host/TestPlannedParticle.kt

```diff
@@ -0,0 +1,19 @@
+package arcs.core.host
+
+import arcs.core.data.Plan
+import arcs.core.data.Schema
+import arcs.core.entity.EntityBaseSpec
+import arcs.sdk.BaseParticle
+import arcs.sdk.HandleHolderBase
+
+class TestPlannedParticle(val spec: Plan.Particle?) : BaseParticle() {
+    val schema = spec?.handles?.getValue("data")?.type?.toSchema() ?: Schema.EMPTY
+    override val handles = Handles(schema)
+
+    class Handles(val schema: Schema) : HandleHolderBase(
+        "TestConstructedParticle",
+        mapOf(
+            (schema.name?.name ?: "anonymousSchema") to EntityBaseSpec(schema)
```

Comment:
Is this a map from a connection name? So, it should be:

```
mapOf(
  "data" to EntityBaseSpec(schema)
)
```

I think.

Is it possible to verify in a test that you actually get the schema handed over?

Reply:
Yes -- that's a good point (probably make before in these comments) that there should be more tests for these dynamic particles. After I get everything building, I'll add those tests to this branch. 

Clarifying question: For "planned" or "reflective" particles (particles that accept `Plan.Particle`), what is in scope for being dynamic? Given your suggestion, it seems like handle connection names should be rigid, but their types dynamic. If that's the case, how would a particle know of the connection name in advance?

Reply:
Yes, exactly as you said - list of connections and their names are set, it's only types that are dynamic.